sdk/identity/azure-identity/azure/identity/_credentials/service_fabric.py

```diff
@@ -0,0 +1,77 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import functools
+import os
+from typing import TYPE_CHECKING
+
+from azure.core.credentials import AccessToken
+from azure.core.pipeline.transport import HttpRequest
+
+from .. import CredentialUnavailableError
+from .._constants import EnvironmentVariables
+from .._internal.managed_identity_client import ManagedIdentityClient
+from .._internal.get_token_mixin import GetTokenMixin
+
+if TYPE_CHECKING:
+    from typing import Any, Optional
+
+
+class ServiceFabricCredential(GetTokenMixin):
+    def __init__(self, **kwargs):
+        # type: (**Any) -> None
+        super(ServiceFabricCredential, self).__init__()
+
+        client_args = _get_client_args(**kwargs)
+        if client_args:
+            self._client = ManagedIdentityClient(**client_args)
+        else:
+            self._client = None
+
+    def get_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        if not self._client:
+            raise CredentialUnavailableError(
+                message="Service Fabric managed identity configuration not found in environment"
+            )
+        return super(ServiceFabricCredential, self).get_token(*scopes, **kwargs)
+
+    def _acquire_token_silently(self, *scopes):
+        # type: (*str) -> Optional[AccessToken]
+        return self._client.get_cached_token(*scopes)
+
+    def _request_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        return self._client.request_token(*scopes, **kwargs)
+
+
+def _get_client_args(**kwargs):
+    # type: (dict) -> Optional[dict]
+    identity_config = kwargs.pop("identity_config", None) or {}
+
+    url = os.environ.get(EnvironmentVariables.IDENTITY_ENDPOINT)
+    secret = os.environ.get(EnvironmentVariables.IDENTITY_HEADER)
+    thumbprint = os.environ.get(EnvironmentVariables.IDENTITY_SERVER_THUMBPRINT)
+    if url and secret and thumbprint:
+        version = "2019-07-01-preview"
+        base_headers = {"Secret": secret}
+        connection_verify = False
+    else:
+        # Service Fabric managed identity isn't available in this environment
+        return None
+
+    return dict(
+        kwargs,
+        _identity_config=identity_config,
+        base_headers=base_headers,
+        connection_verify=connection_verify,
+        request_factory=functools.partial(_get_request, url, version),
+    )
+
+
+def _get_request(url, version, scope, identity_config):
+    # type: (str, str, str, dict) -> HttpRequest
+    request = HttpRequest("GET", url)
+    request.format_parameters(dict({"api-version": version, "resource": scope}, **identity_config))
```

Comment:
The doc says that "an application can obtain an access token representing its identity, which may be system-assigned or user-assigned, and use it as a 'bearer' token to authenticate itself to another service", so it looks like the request doesn't need to specify a configuration. Thanks for catching this!

Reply:
Shouldn't this then be:
```suggestion
    request.format_parameters({"api-version": version, "resource": scope})
```

Reply:
Another thought: if the endpoint doesn't allow selecting identities at runtime via request parameters, passing `client_id` to this credential probably ought to be a `ValueError`.

Reply:
Would that be a check done in `_get_client_args` (which is where `client_id` is handled for AppServiceCredential)? And would that error just inform the user that they can't select identities with SF MI?

Reply:
I might do something like this in `ServiceFabricCredential.__init__`:
```py
for keyword in ("client_id", "identity_config"):
    if keyword in kwargs:
        message = (
            'The "{}" argument is not accepted because Service Fabric'.format(keyword)
            + "doesn't allow applications to dynamically select user-assigned identities"
        )
        raise ValueError(message)
```
but I'd want to be sure the statement's true, and know whether the endpoint responds with an error message that's just as good when it gets a request having one of the usual user-assigned identity parameters.

Reply:
To resolve this: after discussing with the service team, they want us to pass along any `client_id` we get from the user and they'll handle any incorrect values on their side. User-assigned identity is supported: an app can have multiple identities assigned to it, but each service in the app can only have one identity.